game-app/game-core/src/main/java/games/strategy/triplea/attachments/RulesAttachment.java

```diff
@@ -1001,6 +1072,14 @@ private boolean matchTerritories(
     return numberMet >= getTerritoryCount();
   }
 
+  private boolean checkCheckAIPlayer(final List<GamePlayer> players) {
+    boolean bcheck = true;
+    for (GamePlayer player : players) {
+      bcheck = false ? false : player.isAi();
```

Comment:
This doesn't look right. The "false ? false" part does nothing, so the code is the same as:

```bcheck = player.isAi();```

So that means it will only check the last player in the list.
I assume you want to it either check that it's true for all the players or for at least one. Do you know if there's a precedent for such conditions for which one it should be doing?

Reply:
The pass-through value is "false". In this case if the map maker wishes to use the "players" option, to check if a 'team', "Americans" and "British" were both AI controlled. If one 'team' member is not, then the "false" value must be maintained. Thus if bcheck is false, it must stay false. 
~~~
<option name="checkAIPlayer" value="false"/>
~~~
Also does nothing.

Hope this explains the logic.

Cheers...

Reply:
This code isn't checking for the value "false" though.

You're not doing something like `someValue.equals("false")`.

```
bcheck = false ? false : player.isAi();
```
Is the same as:
```
if (false) {
  bcheck = false;
} else {
  bcheck = player.isAi();
}
```

And the above is the same as:
```
bcheck = player.isAi();
```

Reply:
> to check if a 'team', "Americans" and "British" were both AI controlled

So that sounds like you want to check that *both* players are AI controlled.
Please document that with a comment.

I think your code was supposed to do:
```bcheck = (bcheck == false) ? false : player.isAi();```

Which btw is the same as this shorter version:

```bcheck = bcheck && player.isAi();```

Reply:
You are right!  The line should be changed. Would I need to remove this merge and submit a new one to do this correction.

Reply:
Line changed to read 
`bcheck = !bcheck ? false : player.isAi()`
because if both bcheck and player.isAi() are false then 
`bcheck = bcheck && player.isAi()` would equal true which is not what is wanted.

Reply:
If bcheck and player.isAi() are false, then the result will be false.

(false && false) == false

Reply:
Really? I always thought false and false equals true. But we are saying "and" here and not equals, so that may be what is confusing to me.

Am creating testing for RulesAttachment. 

Cheers...